python/tvm/target/virtual_device.py

```diff
@@ -27,7 +27,9 @@ class VirtualDevice(Object):
     """A compile time representation for where data is to be stored at runtime,
     and how to compile code to compute it."""
 
-    def __init__(self, device, target=None, memory_scope="") -> None:
+    def __init__(self, device=None, target=None, memory_scope="") -> None:
+        if device is None:
+            device = tvm.device(-1, -1)
```

Comment:
Added a comment. Yes, it just means 'nothing known about the device type or id, it may be filled in during device planning based on how the contraints work out'.